src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -163,6 +164,27 @@ public class EC2FleetCloud extends AbstractEC2FleetCloud {
     private transient AtomicInteger plannedNodeCounter = new AtomicInteger(1);
 
     @DataBoundConstructor
+    public EC2FleetCloud(@Nonnull final String name,
+                         final @Deprecated String credentialsId,
+                         final String region,
+                         final String fleet,
+                         final String labelString,
+                         final ComputerConnector computerConnector) {
+        super(StringUtils.isNotBlank(name) ? name : CloudNames.generateUnique(BASE_DEFAULT_FLEET_CLOUD_ID));
+        init();
+        this.credentialsId = credentialsId;
+        this.region = region;
+        this.fleet = fleet;
+        this.computerConnector = computerConnector;
+        this.labelString = labelString;
+
+        if (fleet != null) {
+            this.stats = EC2Fleets.get(fleet).getState(
+                    getAwsCredentialsId(), region, endpoint, getFleet());
+        }
+    }
+
+    @Deprecated
```

Comment:
Lets reuse the new constructor in the deprecated one via constructor chaining to avoid duplicated code. 
Also, should the deprecated constructor use the setters so the validations and defaults are consistent?

Reply:
Yup, it seems like this is what we should do based on the [Kubernetes Plugin Cloud](https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java).